src/register/register-gnome/gnucash-sheet-private.c

```diff
@@ -699,19 +699,28 @@ gnucash_sheet_draw_cursor (GnucashCursor *cursor, cairo_t *cr)
 
     fg_color = &gn_black;
 
-    /* draw the rectangle around the entire active
-     *    virtual *row */
-    cairo_rectangle (cr, cursor->x - x + 0.5, cursor->y - y + 0.5,
-                     cursor->w - 1.0, cursor->h - 1.0);
-    cairo_move_to (cr, cursor->x - x, cursor->y - y + cursor->h - 1.5);
-    cairo_rel_line_to (cr, cursor->w, 0);
     cairo_set_source_rgb (cr, fg_color->red, fg_color->green, fg_color->blue);
+    /* draw the rectangle around the entire active virtual row - transaction rows only - double line  */
+
+    if (cursor->y == 0)
+        cairo_rectangle (cr, cursor->x - x + 0.5, cursor->y - y + 0.5, cursor->w - 1.0, cursor->h - 3.0);
+    else
+    {
+        if (cursor->x == 0)
+            cairo_rectangle (cr, cursor->x - x + 0.5, cursor->y - y - 0.5, cursor->w - 1.0, cursor->h - 2.0);
+        else
+            cairo_rectangle (cr, cursor->x - x - 0.5, cursor->y - y - 0.5, cursor->w, cursor->h - 2.0);
+    }
```

Comment:
It seems your desire to display a frame border on first row/cell does require quite a bit of extra work, considering this commit, the next and parts of the commit in which you redo all the border drawing.
How about instead keeping the borders as they were and draw one black frame around the whole visible area of the gtklayout ? That would avoid all the exceptions you have to code now at the expense of some redundant lines at the bottom and lower parts of the left and right border if the row don't completely fill the visible area.
A variation would be to calculate the vertical position of the bottom most row and draw the left-most and right-most border in one go at the end. when all rows are drawn.